test/Reader.hs

```diff
@@ -11,53 +12,60 @@ import qualified Control.Monad.Trans.Reader as ReaderT
 import qualified Control.Monad.Trans.RWS.Lazy as LazyRWST
 import qualified Control.Monad.Trans.RWS.Strict as StrictRWST
 import Data.Function ((&))
-import Data.Functor.Identity (Identity)
 import Gen
+import GHC.Generics ((:.:)(..))
 import qualified Monad
 import qualified MonadFix
 import Test.Tasty
 import Test.Tasty.Hedgehog
 
 tests :: TestTree
 tests = testGroup "Reader"
-  [ testGroup "ReaderC"       $
+  [ testGroup "ReaderC" $
     [ testMonad
     , testMonadFix
     , testReader
-    ] >>= ($ RunR (uncurry ReaderC.runReader))
-  , testGroup "(->)"          $ testReader (RunR (uncurry (fmap PureC . (&))))
-  , testGroup "ReaderT"       $ testReader (RunR (uncurry (flip ReaderT.runReaderT)))
-  , testGroup "RWST (Lazy)"   $ testReader (RunR (uncurry (runRWST LazyRWST.runRWST)))
-  , testGroup "RWST (Strict)" $ testReader (RunR (uncurry (runRWST StrictRWST.runRWST)))
+    ] >>= ($ runR (uncurry ReaderC.runReader . lower))
+  , testGroup "(->)"          $ testReader (runR (uncurry (fmap pure . (&))            . lower))
+  , testGroup "ReaderT"       $ testReader (runR (uncurry (flip ReaderT.runReaderT)    . lower))
+  , testGroup "RWST (Lazy)"   $ testReader (runR (uncurry (runRWST LazyRWST.runRWST)   . lower))
+  , testGroup "RWST (Strict)" $ testReader (runR (uncurry (runRWST StrictRWST.runRWST) . lower))
   ] where
-  testMonad    run = Monad.test    (m (gen r)) a b c (atom "(,)" (,) <*> r <*> unit) run
-  testMonadFix run = MonadFix.test (m (gen r)) a b   (atom "(,)" (,) <*> r <*> unit) run
-  testReader   run = Reader.test r (m (gen r)) a                                     run
+  testMonad    run = Monad.test    (m (gen0 r) (genN r)) a b c (Comp1 <$> (identity <*> (pair <*> r <*> unit))) run
+  testMonadFix run = MonadFix.test (m (gen0 r) (genN r)) a b   (Comp1 <$> (identity <*> (pair <*> r <*> unit))) run
+  testReader   run = Reader.test r (m (gen0 r) (genN r)) a                (identity <*>                 unit)   run
```

Comment:
Huh, I didn't realize that the current module can implicitly be referred to as if it were a qualified import.

Am I understanding that correctly?

Reply:
You are! Basically, every module in scope is implicitly available under its full name unless it’s renamed using `as`. And the current module is always in scope too. I imagine you can even make life difficult for yourself by importing other modules under the current module’s name.